### PR TITLE
Call, Register, Login Endpoints + Database Layer

### PR DESCRIPTION
**Pull request by suyashkumar**
This change implements the following:
* Call, Register, and Login endpoints
* Database infrastructure needed to support the above

Several changes were made in suyashkumar/auth to support the changes made here.

### CODE REVIEW

File: server/db/db.go [hunk 1]
```diff
@@ -1,32 +1,98 @@
...
+	"github.com/suyashkumar/conduit/server/entities"
 )
 
 const DefaultMaxIdleConns = 5
 
-var db *gorm.DB
+var ErrorNoConnectionString = errors.New("A connection string must be specified on the first call to Get")
+
+// DatabaseHandler abstracts away common persistence operations needed for this package
+type DatabaseHandler interface {
```
Comment (suyashkumar) [inline server/db/db.go]: perhaps should just be `Handler` to semtantically read as `db.Handler`

Comment (suyashkumar) [reply server/db/db.go]: this would also match `device.Handler` pattern

File: server/db/db.go [hunk 2]
```diff
@@ -1,32 +1,98 @@
...
+	// GetUser gets a user from the database that matches constraints on the input user
+	GetUser(u auth.User) (auth.User, error)
```
Comment (suyashkumar) [inline server/db/db.go]: I think using interface composition here could be cool

File: server/config/defaults.go [hunk 3]
```diff
@@ -6,10 +6,12 @@ const Port = "Port"
 const CertKey = "CertKey"
 const PrivKey = "PrivKey"
 const UseSSL = "UseSSL"
+const SigningKey = "SigningKey"
 
 var defaults = map[string]string{
 	LogFile:      "logs.txt",
```
Comment (Ouwen) [inline server/config/defaults.go:12]: `LogFilePath` instead?

File: server/db/db.go [hunk 4]
```diff
@@ -1,32 +1,98 @@
...
+	return d.authDBHandler.GetUser(u)
+}
 
-	c := config.Get(config.DBConnString)
-	d, err := gorm.Open("postgres", c)
+func (d *databaseHandler) UpsertUser(u auth.User) error {
+	return d.authDBHandler.UpsertUser(u)
+}
 
+func (d *databaseHandler) GetDeviceSecret(uuid uuid.UUID) (entities.DeviceSecret, error) {
```
Comment (Ouwen) [inline server/db/db.go]: `AccountSecret`

File: server/handlers/api_handlers.go [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+	if err != nil {
+		logrus.WithError(err).Error("Unknown error validating token")
+		err := sendJSON(w, entities.ErrorResponse{Error: "Error validating token"}, 500)
+		if err != nil {
+			logrus.WithError(err).Error("!!!! Could not send error JSON response (CallRequest)")
+		}
+		return
+	}
+
+	d.Call(req.DeviceName, claims.Data["deviceSecret"], req.FunctionName)
```
Comment (Ouwen) [inline server/handlers/api_handlers.go]: #59

Comment (suyashkumar) [reply server/handlers/api_handlers.go]: Sounds good for v2.1! v2 will be focused on feature parity with v1!

**Pull request merged**